src/lib/network_pool/transaction_pool.ml

```diff
@@ -1225,10 +1225,13 @@ struct
                 return
                   { diff with
                     data =
-                      List.map commands
-                        ~f:
+                      List.map commands ~f:(fun ((cmd, _) as cmd_with_keys) ->
+                          let hash =
+                            Transaction_hash.hash_command
+                              (User_command.forget_check cmd)
+                          in
                           Transaction_hash.User_command_with_valid_signature
-                          .create
```

Comment:
Currently `create` assumes keys list from `cmd_with_keys` to be empty. I suggest that we introduce a `create_with_keys` function that would take `cmd_with_keys` parameter and compute transaction hash inside, instead of using `make`.